include/xleaflet/test_xleaflet.hpp

```diff
@@ -0,0 +1,199 @@
+            {
+                return obj.cast<long>();
+            }
+            if (py::isinstance<py::float_>(obj))
+            {
+                return obj.cast<double>();
+            }
+            if (py::isinstance<py::str>(obj))
+            {
+                return obj.cast<std::string>();
+            }
+            if (py::isinstance<py::tuple>(obj) || py::isinstance<py::list>(obj))
+            {
+                auto out = json::array();
+                for (const py::handle& value : obj)
+                {
+                    out.push_back(to_json_impl(value));
+                }
+                return out;
+            }
+            if (py::isinstance<py::dict>(obj))
+            {
+                auto out = json::object();
+                for (const py::handle& key : obj)
+                {
+                    out[py::str(key).cast<std::string>()] = to_json_impl(obj[key]);
+                }
+                return out;
+            }
+            throw std::runtime_error("to_json not implemented for this type of object: " + obj.cast<std::string>());
+        }
+    }
+
+    py::object adl_serializer<py::object>::from_json(const json& j)
+    {
+        return detail::from_json_impl(j);
+    }
+
+    void adl_serializer<py::object>::to_json(json& j, const py::object& obj)
+    {
+        j = detail::to_json_impl(obj);
+    }
+}
+
+std::set<std::string> state_set(auto xleaflet_widget)
+{
+    xeus::xjson state; xeus::buffer_sequence buffers;
+    xleaflet_widget.serialize_state(state, buffers);
+    std::set<std::string> state_set;
+    for (auto& el : state.items()){
+        state_set.insert(el.key());
+    }
+    return state_set;
+}
+
+std::vector<std::string> intersection_widget(std::set<std::string> parsed_set, std::set<std::string> state_set)
+{
+    std::vector<std::string> intersection;
+    std::set_intersection(begin(parsed_set), end(parsed_set),
+                          begin(state_set), end(state_set),
+                          std::back_inserter(intersection));
+
+    std::vector<std::string>::iterator it = intersection.begin();
+
+    while(it != intersection.end())
+    {
+        std::cout << *it << " \n";
+        it++;
+    }
+    std::cout<<std::endl;
+    return intersection;
+}
+
+std::vector<std::string> difference_widget(std::set<std::string> parsed_set, std::set<std::string> state_set)
+{
+    std::cout << "\nAttributes present in ipyleaflet not in xleaflet : " << "\n" << std::endl;
+
+    std::vector<std::string> difference;
+    std::set_difference(begin(parsed_set), end(parsed_set),
+                        begin(state_set), end(state_set),
+                        std::back_inserter(difference));
+    std::vector<std::string>::iterator it = difference.begin();
+    while(it != difference.end())
+    {
+        std::cout << *it << " \n";
+        it++;
+    }
+    std::cout<<std::endl;
+    return difference;
+}
+
+std::set<std::string> set_item(xeus::xjson& item_set)
```

Comment:
`std::set<std::string> extract_attribute_names(xeus::xjson& widget_representation)` would be clearer